redesign post-list

aliceriot opened the pull request:
#### What are the relevant tickets?

closes #776 

#### What's this PR do?

Implements a redesign of the post list, which we use on the channel page and homepage and on the channel moderation page.

#### How should this be manually tested?

It should look alright and implement the things the design is supposed to have!

Also, check that all the functionality we currently offer on those pages still works alright.

noisecapella commented:
I know this is one of those unsolved problems in computing but the select doesn't look right on firefox: 
![screenshot from 2018-06-25 16-29-29](https://user-images.githubusercontent.com/863262/41874145-00ff85a0-7895-11e8-9065-2d8d37e98fca.png)

noisecapella commented:
In the mockup is this supposed to be the icon for a text post? cc @roberthouse54 

![screenshot from 2018-06-25 16-36-49](https://user-images.githubusercontent.com/863262/41874465-02238958-7896-11e8-93c2-89e93dd9404f.png)

noisecapella commented:
Functionality looks great! I'll look over code and tests tomorrow

noisecapella commented on static/js/components/UserMenu_test.js:
```diff
@@ -62,12 +59,14 @@ describe("UserMenu", () => {
     })
   })
+
+  //
```
Stray comment

aliceriot replied:
yeah, I put these in intentionally to force prettier to leave a blank line between tests declare with an array, just cause it annoys me, haha

noisecapella commented on static/js/components/UserMenu_test.js:
```diff
@@ -93,6 +93,8 @@ describe("UserMenu", () => {
     assert.isNotOk(wrapper.find(".profile-incomplete").exists())
   })
+
+  //
```
Stray comment

noisecapella commented on static/js/containers/ChannelPage.js:
```diff
@@ -144,17 +143,14 @@ class ChannelPage extends React.Component<*, void> {
-              </div>
-            }
-          >
+          <div>
```
Would this be a good place to use `React.Fragment` instead?

aliceriot replied:
yeah, good idea! The endless battle against extra `<div>` tags

noisecapella commented:
One other thing, the dropdown is empty for anonymous users because they can't do any of the actions. Should we remove the button in that case? Otherwise it looks a little strange:
![screenshot from 2018-06-26 14-48-05](https://user-images.githubusercontent.com/863262/41932861-3a0db7fc-7950-11e8-97b6-8f022aa50504.png)

aliceriot commented:
ok, so I removed the menu button for anonymous users and switched over to react.fragment in that spot. I can't find where Lucas got the icon for the 'text' post from, so I think we can just leave it off for now and when he lets me know I can add it. As for firefox :woman_shrugging: I googled a little but didn't find anything super helpful

aliceriot commented:
@noisecapella I just added a tiny change, if you want to take a quick look

noisecapella commented:
Finally got my laptop set up to test this. Looks great! :+1:
